shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -1024,7 +1016,17 @@ trait StatementImpl extends Located { self: Statement =>
     case LetS(isRec, name, rhs) => s"let${if (isRec) " rec" else ""} ${name.showDbg} = ${rhs.showDbg}"
     case DatatypeDefn(head, body) => s"data type ${head.showDbg} of ${body.showDbg}"
     case DataDefn(head) => s"data ${head.showDbg}"
-    case _: Constructor | _: Term | _: Decl | _: NuDecl => ??? // Implemented in subclasses
+    case Constructor(params, body) => s"constructor(${params.showElems}) ${body.showDbg}"
+    case t: Term => t.print(false)
+    case d: Decl => d.showHead + (d match {
+      case TypeDef(Als, _, _, _, _, _, _, _) => " = ";
```

Comment:
This underscore fest is actually an anti-pattern (I know it's used in a few places currently). Just use `case td: TypeDef if td.kind is Als`. Same remark for code below.

Reply:
Ah, I also felt uncomfortable when I saw this, but I ignored it. Now it has been fixed.